store/src/file_sink.rs

```diff
@@ -295,3 +295,70 @@ impl FileSink {
         }
     }
 }
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use async_compression::tokio::bufread::GzipDecoder;
+    use futures::stream::StreamExt;
+    use tokio::{fs::DirEntry, io::BufReader};
+    use tokio_util::codec::{length_delimited::LengthDelimitedCodec, FramedRead};
+
+    #[tokio::test(flavor = "multi_thread", worker_threads = 2)]
+    async fn writes_a_framed_gzip_encoded_file() {
+        let tmp_dir = tempdir::TempDir::new("entropy_tests").expect("Unable to create temp dir");
+        let (shutdown_trigger, shutdown_listener) = triggered::trigger();
+        let (sender, receiver) = message_channel(10);
+
+        let mut file_sink = FileSinkBuilder::new(FileType::Entropy, &tmp_dir.path(), receiver)
+            .roll_time(chrono::Duration::milliseconds(100))
+            .create()
+            .await
+            .expect("failed to create file sink");
+
+        let sink_thread = tokio::spawn(async move {
+            file_sink
+                .run(&shutdown_listener)
+                .await
+                .expect("failed to complete file sink");
+        });
+
+        sender
+            .try_send(String::into_bytes("hello".to_string()))
+            .expect("failed to send bytes to file sink");
+
+        tokio::time::sleep(time::Duration::from_millis(200)).await;
+
+        shutdown_trigger.trigger();
+        sink_thread.await.expect("file sink did not complete");
+
+        let entropy_file = get_entropy_file(&tmp_dir).await;
+        assert_eq!("hello", read_file(&entropy_file).await);
+    }
+
+    async fn read_file(entry: &DirEntry) -> bytes::BytesMut {
+        let gzip_decoder =
+            GzipDecoder::new(BufReader::new(File::open(entry.path()).await.unwrap()));
+        let mut framed_reader = FramedRead::new(gzip_decoder, LengthDelimitedCodec::new());
+
+        framed_reader.next().await.unwrap().unwrap()
+    }
+
+    async fn get_entropy_file(tmp_dir: &tempdir::TempDir) -> DirEntry {
+        let mut entries = fs::read_dir(tmp_dir.path())
+            .await
+            .expect("failed to read tmp dir");
+
+        while let Some(entry) = entries.next_entry().await.unwrap() {
+            if is_entropy_file(&entry) {
+                return entry;
+            }
+        }
+
+        panic!("no entropy file available")
+    }
+
+    fn is_entropy_file(entry: &DirEntry) -> bool {
+        entry.file_name().to_str().unwrap().starts_with("entropy.")
```

Comment:
use `FileType` to parse and then compare with `FileType::Entropy`

Reply:
Addressed in 598a7d7